Automattic/wp-calypso
# Odyssey Stats: Migrate non-component code to TypeScript - Migration Part 1

Related to #76570 

## Proposed Changes

Split off #76619

* Adds `tsconfig.json`
* Adds `tsconfig-reference.json` in `client` to make the project reference-able.
* Migrated all non-component code to TypeScript except for `apps/odyssey-stats/src/lib/create-odyssey-config.js` which would need some major refactoring

## Testing Instructions

* Ensure all checks successful on GitHub
* Build it locally
* Ensure it still builds and works well
* Ensure no TS error reported in VSCode or the editor you are using

## Pre-merge Checklist
